classes/order/OrderSlip.php

```diff
@@ -322,7 +321,7 @@ public static function create(Order $order, $product_list, $shipping_cost = fals
                 continue;
             }
 
-            if (!Tools::isSubmit('cancelProduct') && $order->hasBeenPaid()) {
+            if ($order->hasBeenPaid()) {
```

Comment:
I removed it because this is a dependency to the legacy form (the cancel product form) that won't exist anymore. 

I just added a comment with a todo, since I'm the one who works on the cancel product feature, I'll deal with it in the corresponding pull request 👍

Reply:
Although I agree this should be managed better than this, but it a module uses the legacy page this could change its behaviour I'd say it would be safer to leave it unchanged in the legacy class But if you want to clean this code you extract it in a dedicated service that you can use in your handler Then you'll be free to make the improvements you want